zephyr/wrapper.c

```diff
@@ -90,9 +95,31 @@ static void heap_free(struct k_heap *h, void *mem)
 	k_spin_unlock(&h->lock, key);
 }
 
+static bool zone_is_cached(enum mem_zone zone)
```

Comment:
this should be inline, and I would image most compilers would make it so.

Reply:
yes, I'd not want to use a compiler that wouldn't do that. It's also only used once. So should be fine to let the compiler figure it out.

Reply:
FWIW: whenever I've actually had to sit down and start counting cycles, there are always spots where I disagree with inlining decisions.  For cases where you really really care[1], Zephyr abstracts an ALWAYS_INLINE attribute (which IIRC is only supported in gcc attribute syntax anyway).

[1] Also in the extremely rare case where you want C syntax but where an actual function call would break the semantics.  We have a few of these wrappers around inline assembly that mucks with the stack pointer, for instance, though the only one I can actually remember (the arch_switch() implementation for x86_64) got removed and replaced a while back.

Reply:
In every (toy) example I tried `inline` was totally ignored by the compiler. Only the `-On` optimization level made a (big) difference. Every serious documentation or article I've read concurs, not sure why this manual inlining myth seems to persist.

The only value of `static inline` is apparently to silence `unused function` warnings in the .c files that #include but don't use that function?

Reply:
Jury is out, inline is out.